backend/src/middleware/filterBubble/filterBubble.spec.js

```diff
@@ -26,24 +32,42 @@ const randomAuthorParams = {
 const categoryIds = ['cat9']
 
 beforeEach(async () => {
-  await Promise.all([
+  const [currentUser, followedAuthor, randomAuthor] = await Promise.all([
     factory.create('User', currentUserParams),
     factory.create('User', followedAuthorParams),
     factory.create('User', randomAuthorParams),
   ])
-  await instance.create('Category', {
+  user = currentUser
+  await neode.create('Category', {
     id: 'cat9',
     name: 'Democracy & Politics',
     icon: 'university',
   })
-  const [asYourself, asFollowedUser, asSomeoneElse] = await Promise.all([
-    Factory().authenticateAs(currentUserParams),
-    Factory().authenticateAs(followedAuthorParams),
-    Factory().authenticateAs(randomAuthorParams),
-  ])
-  await asYourself.follow({ id: 'u2', type: 'User' })
-  await asFollowedUser.create('Post', { title: 'This is the post of a followed user', categoryIds })
-  await asSomeoneElse.create('Post', { title: 'This is some random post', categoryIds })
+  await currentUser.relateTo(followedAuthor, 'following')
+  await factory.create('Post', {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T09:16:22Z" title="Tuesday, September 3rd 2019, 11:16:22 am +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -26,24 +32,42 @@ const randomAuthorParams = {
 const categoryIds = ['cat9']
 
 beforeEach(async () => {
-  await Promise.all([
+  const [currentUser, followedAuthor, randomAuthor] = await Promise.all([
     factory.create('User', currentUserParams),
     factory.create('User', followedAuthorParams),
     factory.create('User', randomAuthorParams),
   ])
-  await instance.create('Category', {
+  user = currentUser
+  await neode.create('Category', {
     id: 'cat9',
     name: 'Democracy & Politics',
     icon: 'university',
   })
-  const [asYourself, asFollowedUser, asSomeoneElse] = await Promise.all([
-    Factory().authenticateAs(currentUserParams),
-    Factory().authenticateAs(followedAuthorParams),
-    Factory().authenticateAs(randomAuthorParams),
-  ])
-  await asYourself.follow({ id: 'u2', type: 'User' })
-  await asFollowedUser.create('Post', { title: 'This is the post of a followed user', categoryIds })
-  await asSomeoneElse.create('Post', { title: 'This is some random post', categoryIds })
+  await currentUser.relateTo(followedAuthor, 'following')
+  await factory.create('Post', {
```


One question I have is why are we using `neode` to create certain resources and `factory` to create others?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-03T12:33:35Z" title="Tuesday, September 3rd 2019, 2:33:35 pm +02:00">Sep 3, 2019</time>_

---


Beause we don't have https://github.com/thoughtbot/factory_bot :cry: Our custom factories are just a bad copy of this really sophisticated framework, delivering quality since 2008. If we use `neode` for test setup, it got there before there was a corresponding `factory` for that resource. Without our own factories, however, we might setup a state in the database that wouldn't exist, e.g.posts without categories. I prefer to use factories but I don't really mind if there are traces of `neode` in the test setup.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T13:53:33Z" title="Tuesday, September 3rd 2019, 3:53:33 pm +02:00">Sep 3, 2019</time>_

---


makes sense, I saw some examples where you create a user where you are not concerned in the test that it has all the necessary attributes, just that it has an id